### PR TITLE
docs: rename purity polymorphism to reflection

### LINKED ISSUE #3227: Rename "purity polymorphism / purity polymorphic" to "purity reflection / purity reflective"

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #3227

Depends on #3235

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/errors/CodeHint.scala [hunk 1]
```diff
@@ -66,13 +65,13 @@ object CodeHint {
   }
 
   /**
-    * A code hint that indicates that a purity polymorphic operation is lazy.
+    * A code hint that indicates that a Purity reflective operation is lazy.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/errors/CodeHint.scala:68]: uncapitalize (applies in other places aswell)

**Comment by magnus-madsen** [comment]
Also check "Did you know" document.

**Comment by jaschdoc** [comment]
RFR

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**